test/functional/http/get/delegates.js

```diff
@@ -1,1039 +1,1080 @@
 'use strict';
 
-var node = require('../../../node');
-var http = require('../../../common/httpCommunication');
-var sendLISK = require('../../../common/apiHelpers').sendLISK;
-var sendTransaction = require('../../../common/apiHelpers').sendTransaction;
+var node = require('../../../node.js');
 var modulesLoader = require('../../../common/modulesLoader');
 var genesisDelegates = require('../../../genesisDelegates.json');
-var testDelegate = genesisDelegates.delegates[0];
 
-function getForgingStatus (publicKey, cb) {
-	http.get('/api/delegates/forging/status?publicKey=' + publicKey, function (err, res) {
-		cb(err, res.body);
-	});
-}
-
-//insert one extra delegate
-before(function (done) {
-	var delegate = node.randomAccount();
-	sendLISK({
-		secret: node.gAccount.password,
-		amount: node.randomLISK(),
-		address: delegate.address
-	}, function (err, res) {
-		node.expect(res).to.have.property('success').to.be.ok;
-		node.expect(res).to.have.property('transactionId').that.is.not.empty;
-		node.onNewBlock(function () {
-			var insertDelegateTrs = node.lisk.delegate.createDelegate(delegate.password, delegate.username);
-			sendTransaction(insertDelegateTrs, function (err, res) {
-				node.expect(err).to.be.null;
-				node.onNewBlock(done);
-			});
-		});
-	});
-});
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var registerDelegatePromise = require('../../../common/apiHelpers').registerDelegatePromise;
+var getDelegatesPromise = require('../../../common/apiHelpers').getDelegatesPromise;
+var getCountPromise = require('../../../common/apiHelpers').getCountPromise;
+var getVotersPromise = require('../../../common/apiHelpers').getVotersPromise;
+var getForgingStatusPromise = require('../../../common/apiHelpers').getForgingStatusPromise;
+var searchDelegatesPromise = require('../../../common/apiHelpers').searchDelegatesPromise;
+var putForgingDelegatePromise = require('../../../common/apiHelpers').putForgingDelegatePromise;
+var getForgedByAccountPromise = require('../../../common/apiHelpers').getForgedByAccountPromise;
+var getNextForgersPromise = require('../../../common/apiHelpers').getNextForgersPromise;
+var onNewBlockPromise = node.Promise.promisify(node.onNewBlock);
+var onNewRoundPromise = node.Promise.promisify(node.onNewRound);
 
-describe('GET /api/delegates (cache)', function () {
-	var cache;
+describe('GET /api/delegates', function () {
 
-	before(function (done) {
-		node.config.cacheEnabled = true;
-		modulesLoader.initCache(function (err, __cache) {
-			cache = __cache;
-			node.expect(err).to.not.exist;
-			node.expect(__cache).to.be.an('object');
-			return done(err, __cache);
+	var testDelegate = genesisDelegates.delegates[0];
+	
+	describe('from (cache)', function () {
```

Comment:
Can `describe` have the `@slow` badge as well?

Reply:
Yes, It can. But it will affect all the tests inside that describe block.